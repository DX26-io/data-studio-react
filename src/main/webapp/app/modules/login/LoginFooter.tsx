import React from 'react';
import { Text } from '@adobe/react-spectrum'

import { CommonConstants } from '../../shared/util/common.constants';

// TODO: Test Cases
// TODO : year should come dynamically 

const LoginFooter = props => (
    <Text position="absolute" right="3%" bottom="4%">{CommonConstants.DX26} {CommonConstants.COPYRIGHT} 2020</Text>
);

export default LoginFooter;
