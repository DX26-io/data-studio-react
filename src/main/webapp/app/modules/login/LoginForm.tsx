import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { TextField, Flex, Form, Checkbox, Button, Text } from '@adobe/react-spectrum';
import Typography from '@material-ui/core/Typography';

export interface ILoginProps {
  loginError: boolean;
  handleLogin: (username: string,password: string, rememberMe: boolean) => void;
}

export const LoginForm = (props: ILoginProps) => {

  const [username, setUserName] = useState('');
  const [password, setPassword] = useState('');
  const [rememberMe, setRememberMe] = useState(false);

  const handleSubmit = (event) => {
    event.preventDefault();
    const { handleLogin } = props;
    handleLogin(username, password, rememberMe);
  };

  const { loginError } = props;

  return (
    <Form onSubmit={handleSubmit} aria-label="Login Form" width="50%" margin="auto" marginTop="33%">
      <Typography className="dx26-font-bold" variant="h4">Login</Typography>
      <br />
      <TextField label="Email Address" isQuiet isRequired type="text" value={username} onChange={setUserName} />
      <TextField label="Password" isQuiet isRequired type="password" value={password} onChange={setPassword} />
      <Checkbox isSelected={rememberMe} onChange={setRememberMe}>Remember me</Checkbox>
      <br />
      <Flex direction="row" alignItems="center" justifyContent="center">
        <Text>Sign In using&nbsp;
        <Link to="/" className="dx26-link">
            SSO
        </Link>
        </Text>
        <Button variant="cta" marginStart="auto" type="submit">Sign In</Button>
      </Flex>
    </Form>
  );
};

export default LoginForm;
