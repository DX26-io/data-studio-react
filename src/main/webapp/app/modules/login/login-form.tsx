import React, { useState } from 'react';
import { Button, Checkbox, Flex, Text, TextField, View, Link, Form } from '@adobe/react-spectrum';
import Alert from '@spectrum-icons/workflow/Alert';
import { Translate } from 'react-jhipster';
import config from "app/config/constants";

export interface ILoginProps {
  loginError: boolean;
  handleLogin: (username: string, password: string, rememberMe: boolean) => void;
  handleProviderLogin: (provider: string) => void;
  handleSignup: () => void;
}
export const LoginForm = (props: ILoginProps) => {
  const firebaseEnabled = config.CLOUD;
  const [username, setUserName] = useState('');
  const [password, setPassword] = useState('');
  const [rememberMe, setRememberMe] = useState(false);
  const [emptyFieldError, setEmptyFieldError] = useState(false);
  const { handleLogin } = props;
  const { handleProviderLogin } = props;
  const { handleSignup } = props;
  const { loginError } = props;

  const onGoogleClick = () => {
    handleProviderLogin('google');
  }

  const onGitHubClick = () => {
    handleProviderLogin('github');
  }

  const onSignup = () => {
    handleSignup();
  }

  const handleSubmit = event => {
    event.preventDefault();
    if (username && password) {
      setEmptyFieldError(false);
      handleLogin(username, password, rememberMe);
    } else {
      setEmptyFieldError(true);
    }
  };

  return (
    <Flex alignItems="center" justifyContent="center" marginX="size-300">
      <Form width="size-4600" isRequired isQuiet onSubmit={handleSubmit} data-testid="login-form">
        <View marginBottom="size-600">
          <span className="spectrum-Heading spectrum-Heading--sizeXXL">
            <Text>
              <Translate contentKey="login.title">Login</Translate>
            </Text>
          </span>
        </View>
        <TextField
          width="100%"
          marginBottom="size-300"
          label="Email Address"
          type="text"
          data-testid="username"
          value={username}
          onChange={setUserName}
        />
        <TextField
          width="100%"
          marginBottom="size-300"
          label="Password"
          type="password"
          data-testid="password"
          value={password}
          onChange={setPassword}
        />
        {loginError && !emptyFieldError && (
          <Flex gap="size-100" data-testid="login-error">
            <Alert color="negative" />
            <Text marginBottom="size-300">
              <span className="spectrum-Body-emphasis error-message">
                <Translate contentKey="login.messages.error.authentication">Your username or password does not match any account</Translate>
              </span>
            </Text>
          </Flex>
        )}
        {emptyFieldError && (
          <Flex gap="size-100" data-testid="empty-error">
            <Alert color="negative" />
            <Text marginBottom="size-300">
              <span className="spectrum-Body-emphasis error-message">
                <Translate contentKey="login.messages.error.emptyUsernameOrPassword"> Username or password should not be empty</Translate>
              </span>
            </Text>
          </Flex>
        )}
        <Checkbox marginTop="static-size-25" isRequired={false} isSelected={rememberMe} onChange={setRememberMe} data-testid="remember-me">
          <Translate contentKey="login.form.rememberMe">Remember me</Translate>
        </Checkbox>
        <Flex data-testid="login-action" marginTop="size-400" alignItems="center" justifyContent="center" direction="row-reverse">
          <Button data-testid="submit" variant="cta" marginStart="auto" type="submit">
            <Translate contentKey="login.form.button">Sign In</Translate>
          </Button>
          {
            firebaseEnabled ?
              <Button data-testid="submit" variant="secondary" marginStart="auto" type="button" onPress={onGoogleClick}>
                <Translate contentKey="login.form.google">Google</Translate>
              </Button>
            :
            null
          }
          {
            firebaseEnabled ?
              <Button data-testid="submit" variant="secondary" marginStart="auto" type="button" onPress={onGitHubClick}>
                <Translate contentKey="login.form.github">GitHub</Translate>
              </Button>
              :
              null
          }
          <Link isQuiet={true}>
            <a href="/">
              <Translate contentKey="login.form.sso">Use Single Sign On (SSO)</Translate>
            </a>
          </Link>
          <Button data-testid="signup" variant="secondary" marginStart="auto" type="button" onPress={onSignup}>
            <Translate contentKey="login.form.signup">Sign up</Translate>
          </Button>
        </Flex>
      </Form>
    </Flex>
  );
};

export default LoginForm;
