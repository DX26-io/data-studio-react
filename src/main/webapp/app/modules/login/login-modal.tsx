import React from 'react';
import { Provider, defaultTheme, Form, TextField, Checkbox, Button } from '@adobe/react-spectrum';

export interface ILoginModalProps {
  showModal: boolean;
  loginError: boolean;
  handleLogin: Function;
  handleClose: Function;
}

class LoginModal extends React.Component<ILoginModalProps> {
  handleSubmit = (event, errors, { username, password, rememberMe }) => {
    const { handleLogin } = this.props;
    handleLogin(username, password, rememberMe);
  };

  render() {
    return (
      <Provider theme={defaultTheme}>
        <Form maxWidth="size-3600">
          <TextField label="Email" isRequired placeholder="Email Addess" />
          <TextField label="Password" isRequired type="password" placeholder="Password" />
          <Checkbox>Remember me</Checkbox>
          <Button maxWidth="size-200" variant="cta">Login</Button>
        </Form>
      </Provider>
    );
  }
}

export default LoginModal;
