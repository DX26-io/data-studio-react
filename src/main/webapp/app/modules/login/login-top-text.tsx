import React from 'react';
import Typography from '@material-ui/core/Typography';

// TODO change this to logo
const LoginTopText = props => (
  <Typography variant="h6" className="dx26-font-bold" style={{ position: 'absolute', right: '3%', top: '4%' }}>
    DX26
  </Typography>
);

export default LoginTopText;
