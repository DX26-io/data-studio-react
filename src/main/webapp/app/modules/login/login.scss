body {
  overflow: hidden;
}
.login-form {
  display: flex;
  flex-direction: column;
  width: 50%;
  margin: auto;
  margin-top: 33%;
}
