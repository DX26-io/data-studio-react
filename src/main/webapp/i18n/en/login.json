{
  "login": {
    "title": "Login",
    "form": {
      "password": "Password",
      "password.placeholder": "Your password",
      "rememberMe": "Remember me",
      "button": "Sign in",
      "google": "Google",
      "github": "GitHub",
      "sso": "Use Single Sign On (SSO)",
      "signup": "Sign up"
    },
    "messages": {
      "error": {
        "authentication": "<strong>Failed to sign in!</strong> Please check your credentials and try again.",
        "emptyUsernameOrPassword": "Username or password should not be empty"
      }
    },
    "password": {
      "forgot": "Did you forget your password?"
    }
  },
  "signup": {
    "title": "Sign up",
    "done": "Sign up completed successfully. Please check your email to complete verification.",
    "form": {
      "password": "Password",
      "password.placeholder": "Your password",
      "username": "Username",
      "email": "Email",
      "firstname": "First name",
      "lastname": "Last name",
      "button": "Sign up",
      "google": "Google",
      "github": "GitHub"
    },
    "messages": {
      "error": {
        "authentication": "Failed to sign up",
        "emptyFields": "All fields are required"
      }
    }
  },
  "realm": {
    "create": {
      "title": "Create realm",
      "form": {
        "realmName": "Realm name",
        "button": "Create"
      },
      "messages": {
        "error": {
          "create": "Failed to create a realm",
          "emptyFields": "All fields are required"
        }
      }
    }
  },
  "flairbiApp": {
    "realm": {
      "created": "Realm created successfully"
    },
    "features": {
      "created": "Feature created successfully"
    }
  }
}
