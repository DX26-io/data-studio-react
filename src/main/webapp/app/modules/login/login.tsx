import React from 'react';
import { connect } from 'react-redux';
import { Redirect, RouteComponentProps } from 'react-router-dom';
import { Grid, View } from '@adobe/react-spectrum';
import { IRootState } from 'app/shared/reducers';
import { login } from 'app/shared/reducers/authentication';
import { loginWithProvider } from 'app/shared/reducers/authentication';
import LoginForm from './login-form';
import LoginFooter from './login-footer';
import LoginHeader from './login-header';
// TODO create login page image
export interface ILoginProps extends StateProps, DispatchProps, RouteComponentProps<{}> {}

export const Login: React.FC<ILoginProps> = props => {
  const handleLogin = (username, password, rememberMe = false) => {
    props.login(username, password, rememberMe);
  };
  const handleProviderLogin = (provider) => {
    props.loginWithProvider(provider);
  };
  const handleSignup = () => {
    props.history.push('/signup');
  };

  if (props.loginProviderEmailConfirmationToken) {
    return (<Redirect
      to={{
        pathname: '/realm',
      }}
    />);
  }

  if (props.isAuthenticated) {
    return <Redirect
      to={{
        pathname: '/',
        search: props.location.search,
        state: {from: props.location},
      }}
    />;
  }

  return (
    <Grid areas={['image login']} columns={['1fr', '2fr']} rows={['auto']} minHeight={window.innerHeight} data-testid="login-container">
      {/* <Image src="https://i.imgur.com/Z7AzH2c.png" alt="alt-text" objectFit="cover" gridArea="image" />*/}
      <View gridArea="image" backgroundColor="gray-400" />
      <View gridArea="login">
        <LoginHeader />
        <LoginForm handleLogin={handleLogin}
                   loginError={props.loginError}
                   handleProviderLogin={handleProviderLogin}
                   handleSignup={handleSignup}/>
        <LoginFooter />
      </View>
    </Grid>
  );
};

const mapStateToProps = ({ authentication }: IRootState) => ({
  isAuthenticated: authentication.isAuthenticated,
  loginError: authentication.loginError,
  loginProviderEmailConfirmationToken: authentication.loginProviderEmailConfirmationToken,
});

const mapDispatchToProps = { login, loginWithProvider };

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(mapStateToProps, mapDispatchToProps)(Login);
