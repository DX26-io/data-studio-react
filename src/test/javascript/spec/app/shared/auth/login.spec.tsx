import React from 'react';
import { Login, ILoginProps } from 'app/modules/login/login';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Login Container', () => {
  let mountedWrapper;

  const defaultProps = {
    login: jest.fn(),
    getSession: jest.fn(),
  };

  const wrapper = (props: ILoginProps) => {
    if (!mountedWrapper) {
      mountedWrapper = render(<Login {...defaultProps} {...props} />);
    }
    return mountedWrapper;
  };

  beforeEach(() => {
    mountedWrapper = undefined;
  });

  it('Renders login container with default Props', () => {
    const component = wrapper(null);
    expect(component).toBeDefined();
  });

  it('login callback ', () => {
    const component = wrapper(null);
    userEvent.click(component.getByTestId('submit'));
    expect(defaultProps.login.mock.calls.length).toEqual(1);
    expect(defaultProps.getSession.mock.calls.length).toEqual(0);
  });
});
