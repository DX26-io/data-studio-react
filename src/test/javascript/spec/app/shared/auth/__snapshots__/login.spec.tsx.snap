// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container Renders login container with default Props 1`] = `
<ForwardRef($f784fbb881b5b101cd7607722a86$var$Grid)
  areas={
    Array [
      "todo login",
    ]
  }
  columns={
    Array [
      "2fr",
      "2fr",
    ]
  }
  minHeight={768}
  rows={
    Array [
      "auto",
    ]
  }
>
  <ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)
    backgroundColor="blue-600"
    gridArea="todo"
    overflow="hidden"
  />
  <ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)
    backgroundColor="static-white"
    gridArea="login"
  >
    <LoginTopText />
    <Component
      handleLogin={[Function]}
    />
    <LoginFooter />
  </ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)>
</ForwardRef($f784fbb881b5b101cd7607722a86$var$Grid)>
`;
